src/app.ts

```diff
@@ -143,6 +155,23 @@ function resolveDependencies(app: App) {
 
   }
 
+  const charts = new DependencyGraph(Node.of(app)).topology()
+    .filter(x => x instanceof Chart);
+
+  for (const parent of charts) {
+    for (const child of charts) {
+
+      // create an explicit chart dependency from nested chart relationships
+      const parentChart = Chart.of(parent);
+      const childChart = Chart.of(child);
+
+      if (parentChart !== childChart && Node.of(parent).tryFindChild(Node.of(child).id)) {
+        Node.of(parentChart).addDependency(childChart);
+        hasDependantCharts = true;
+      }
+    }
+  }
```

Comment:
Simpler alternative: 

```suggestion
for (const parentChart of charts) {
  for (const childChart of Node.of(parentChart).children.filter(x => x instanceof Chart)) {

    // create an explicit chart dependency from nested chart relationships
    Node.of(parentChart).addDependency(childChart);
    hasDependantCharts = true;
  }
}
```